moto/__init__.py

```diff
@@ -171,22 +172,24 @@ def f(*args, **kwargs):
 mock_sdb = lazy_load(".sdb", "mock_sdb", boto3_name="sdb")
 
 
-def mock_all():
-    dec_names = [
-        d
-        for d in dir(sys.modules["moto"])
-        if d.startswith("mock_")
-        and not d.endswith("_deprecated")
-        and not d == "mock_all"
-    ]
+class mock_all(ContextDecorator):
+    def __init__(self):
+        self.mocks = [
+            globals()[mock]()
+            for mock in dir(sys.modules["moto"])
+            if mock.startswith("mock_")
+            and not mock.endswith("_deprecated")
+            and not mock == "mock_all"
+            and not mock == "mock_xray_client"
```

Comment:
I'm not entirely sure why this is a lazyloaded at the top level. I figure it's probably safe - albeit a little dirty - to ignore it given that `mock_xray` still gets included but I'll defer to your judgement here.